### PR TITLE
Extension delete on recompute

### PR DESCRIPTION
**Pull request by samuelgarcia**
@chrishalcrow : this should fix the issue we discussed when recomputing something all children dependency are not valid anymore and have to deleted. Here a simple implementation of this idea.

### CODE REVIEW

**Comment by chrishalcrow** [comment]
Nice, works for me!

Maybe we could warn the user in some cases? E.g. deleting random spikes deletes everything! Or tell them what's happening:
"{child}, which depends on {parent}, has also being deleted."
?

**Comment by samuelgarcia** [comment]
@alejoe91  : what do you think of warning or print when cascade deleteting on re compute ?

**Comment by zm711** [comment]
For me I mean I wonder what the warning does? Because you're not going to say delete then offer a `y|n` in the terminal right. I think it is probably more important to have strong docstrings and documentation to explain that extensions now have parent-child relations and so deleting the parent necessitates deleting the child, no? It might be nice instead in my opinion to offer a function like `get_all_extensions_relationships` that prints out the dependencies of my current extensions so that I can understand how things interact before I as a user delete them. 

For me the warning only helps if I can stop it when I realize that deleting random_spikes deletes waveforms too... But in this case this is just a print that we are putting as a warning. But that's just my two cents.

**Comment by chrishalcrow** [comment]
I know that I would be confused if I deleted "random_spikes" and everything vanished. I might not realise that the other extensions were deleted when I deleted random_spikes. So the warning highlights this fact.

But yeah, if this was documented well that could work too!

Another idea: now that we have these relationships we could have a "compute_parents" option on the compute() method?

File: src/spikeinterface/core/sortinganalyzer.py [hunk 1]
```diff
@@ -1031,6 +1034,23 @@ def has_extension(self, extension_name: str) -> bool:
...
 _possible_extensions = []
 
+global _extension_children
+_extension_children = {}
+
+
+def get_children(extension_name):
+    # recursive children and so grand children and grand grand children
+    # this is usefull to delete child extension on re compute
+    # for instance recompute the "waveforms" need to delete "template" in ms_before is changed
```
[the remark this answers is not in the document]
Comment (samuelgarcia) [reply src/spikeinterface/core/sortinganalyzer.py:1044]: OK I will do this but this function will be private. I think it is internal no ?

[the remark this answers is not in the document]
Comment (samuelgarcia) [reply src/spikeinterface/core/sortinganalyzer.py]: done

**Comment by samuelgarcia** [comment]
Finally I choose the silently deletion.
Recomputing an extension is a corner case. Unless you are trying parameters.

File: src/spikeinterface/core/sortinganalyzer.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (alejoe91) [reply src/spikeinterface/core/sortinganalyzer.py]: Hopefully there will be a 1.0 by then ahahah

**Review by alejoe91 — APPROVED** [review]

**Pull request merged**